ed25519.go

```diff
@@ -110,6 +110,16 @@ func Verify(publicKey *[PublicKeySize]byte, message []byte, sig *[SignatureSize]
 	if !A.FromBytes(publicKey) {
 		return false
 	}
+	return VerifyUncompressedKey(publicKey, &A, message, sig)
+}
+
+// VerifyUncompressedKey returns true iff sig is a valid signature of message by publicKey.
+// This takes in the uncompressed form of the public key (A) to avoid computing that internally.
+func VerifyUncompressedKey(publicKey *[PublicKeySize]byte, A *edwards25519.ExtendedGroupElement, message []byte, sig *[SignatureSize]byte) bool {
```

Comment:
If we introduce a new public method, we should have a test for that.

Reply:
How would such a test differ from the normal verify though? I would basically be replicating the code inside of Verify, and I'm not sure how useful that is as a test.

Reply:
Tests also serve as a form of documentation here. One can look into the test and see the exact behaviour you just described above. Does that make sense to you?